app/models/search/collection_search_form.rb

```diff
@@ -0,0 +1,113 @@
+class CollectionSearchForm
+  extend ActiveModel::Naming
+  include ActiveModel::Conversion
+  include ActiveModel::Validations
+
+  ATTRIBUTES = [
+    :title,
+    :challenge_type,
+    :signup_open,
+    :unrevealed,
+    :anonymous,
+    :closed,
+    :moderated,
+    :owner_ids,
+    :parent_id,
+    :moderator_ids,
+    :maintainer_id,
+    :signups_open_at,
+    :signups_close_at,
+    :assignments_due_at,
+    :works_reveal_at,
+    :authors_reveal_at,
+    :sort_column,
+    :sort_direction,
+    :page
+  ]
+
+  attr_accessor :options
+
+  ATTRIBUTES.each do |filterable|
+    define_method(filterable) { options[filterable] }
+  end
+
+  def initialize(opts={})
+    @options = opts
+    process_options
+    @searcher = CollectionQuery.new(@options)
+  end
+
+  def process_options
+    @options.delete_if { |k, v| v == "0" || v.blank? }
+    set_sorting
+  end
+
+  def set_sorting
+    @options[:sort_column] ||= default_sort_column
+    @options[:sort_direction] ||= default_sort_direction
+  end
+
+  def persisted?
+    false
+  end
+
+  def summary
```

Comment:
Hmm... is this actually used?

Reply:
Just tested to find out it's not. I'll remove it in the upcoming changes.